app/scripts/metamask-controller.js

```diff
@@ -4130,9 +4130,42 @@ export default class MetamaskController extends EventEmitter {
   }
 
   async addTransaction(txParams, options) {
+    ///: BEGIN:ONLY_INCLUDE_IF(blockaid)
+    let securityAlertResponse;
+
+    try {
+      const request = {
+        method: 'eth_sendTransaction',
+        id: options.actionId,
+        origin: options.origin,
+        params: [
+          {
+            from: txParams.from,
+            to: txParams.to,
+            value: txParams?.value,
+            data: txParams?.data,
+          },
+        ],
+      };
+
+      securityAlertResponse = await this.ppomController.usePPOM(
+        async (ppom) => {
+          return ppom.validateJsonRpc(request);
+        },
+      );
+    } catch (e) {
+      log.debug();
```

Comment:
Do we intend to log the error? So `log.debug(e)`

Reply:
yes, we need to log it.

Reply:
it should be `log.error` I think with error details.

Reply:
Oh this is even better. We can do this. @segun